src/RudderClient.ts

```diff
@@ -92,10 +96,14 @@ async function setup(writeKey: string, configuration: Configuration = {}, option
   }
   validateConfiguration(configuration);
 
-  const config = await configure(writeKey, configuration);
-  logDebug("setup: created config");
-  await bridge.setup(config,options);
-  logDebug("setup: setup completed");
+  // Acquire a lock before calling the setup of Native Modules
+  lock.acquire("lock", async function(done) {
+    const config = await configure(writeKey, configuration);
+    logDebug("setup: created config");
+    await bridge.setup(config,options);
+    logDebug(`setup: setup completed of RudderSDK version ${SDK_VERSION}`);
```

Comment:
Shall we move this out. If setup gets blocked, we won't get to know the version. 

We should log this very early in setup.

Reply:
Yes, will move it